controllers/kokumetricsconfig_controller.go

```diff
@@ -292,6 +305,58 @@ func GetAuthSecret(r *MetricsConfigReconciler, cr *metricscfgv1beta1.MetricsConf
 	return nil
 }
 
+// GetServiceAccountSecret Obtain the client id and client secret from the service account data provided in the current namespace
+func (r *MetricsConfigReconciler) GetServiceAccountSecret(ctx context.Context, cr *metricscfgv1beta1.MetricsConfig, authConfig *crhchttp.AuthConfig, reqNamespace types.NamespacedName) error {
+	log := log.WithName("GetServiceAccountSecret")
+
+	log.Info("secret namespace", "namespace", reqNamespace.Namespace)
+
+	// Fetching the Secret object
+	secret := &corev1.Secret{}
+	secretName := cr.Spec.Authentication.AuthenticationSecretName
+	namespace := types.NamespacedName{
+		Namespace: reqNamespace.Namespace,
+		Name:      secretName}
+	err := r.Get(ctx, namespace, secret)
+	if err != nil {
+		switch {
+		case errors.IsNotFound(err):
+			log.Error(err, "service account secret does not exist")
+		case errors.IsForbidden(err):
+			log.Error(err, "operator does not have permission to check service account secret")
+		default:
+			log.Error(err, "could not check service account secret")
+		}
+		return err
+	}
+
+	// Extracting data from the Secret
+	keys := make(map[string]string)
+	for k, v := range secret.Data {
+		sanitizedKey := sanitizeKey(k)
+		keys[sanitizedKey] = string(v)
+	}
+
+	// Defining the required keys
+	requiredKeys := []string{authClientId, authClientSecret}
+	for _, requiredKey := range requiredKeys {
+		if value, exists := keys[requiredKey]; !exists || value == "" {
```

Comment:
This is a lot of logic when you can just do
```suggestion
		if len(keys[requiredKey]) <= 0 {
```

https://go.dev/play/p/J0JXP9LikO2

Reply:
🤦🏾 indeed, I've updated it